documentation/docs/02-runes/02-$state.md

```diff
@@ -147,6 +147,35 @@ person = {
 
 This can improve performance with large arrays and objects that you weren't planning to mutate anyway, since it avoids the cost of making them reactive. Note that raw state can _contain_ reactive state (for example, a raw array of reactive objects).
 
+## State options
+
+Both `$state` and `$state.raw` can optionally accept a second argument. This allows you to specify an `onchange` function that will be called synchronously whenever the state value changes (for `$state` it will also be called for deep mutations).
+
+The `onchange` function is untracked so even if you assign within an `$effect` it will not cause unwanted dependencies.
+
+```js
+let count = $state(0, {
+	onchange(){
+		console.log("count is now", count);
+	}
+});
+
+// this will log "count is now 1"
+count++;
+```
+
+this could be especially useful if you want to sync some stateful variable that could be mutated without using an effect.
+
+```js
+let array = $state([], {
+	onchange(){
+		localStorage.setItem('array', JSON.stringify(array));
+	}
+});
+
+array.push(array.length);
+```
```

Comment:
even though localStorage was one of the primary motivators for this work I think we need to omit it from these docs, because it's just too hard to illustrate concisely without cutting corners:

- needs to accept initial data _without_ blowing up in SSR (i.e. needs `typeof localStorage` checks etc)
- needs to be composable
- needs to react to `storage` events, which basically means using `createSubscriber` which is a whole thing

I think a simpler and clearer example would be validation. Will push a change